common/_static/helpfulness.js

```diff
@@ -0,0 +1,345 @@
+/* msc-throttle.js - start */
+$.throttle = function (fn, threshhold, scope) {
+	threshhold || (threshhold = 250);
+	var last,
+		deferTimer;
+	return function () {
+		var context = scope || this;
+
+		var now = +new Date,
+			args = arguments;
+		if (last && now < last + threshhold) {
+			// hold on to it
+			clearTimeout(deferTimer);
+			deferTimer = setTimeout(function () {
+				last = now;
+				fn.apply(context, args);
+			}, threshhold);
+		} else {
+			last = now;
+			fn.apply(context, args);
+		}
+	};
+}
+/* msc-throttle.js - end */
+
+/* msc-helpfulness.js - start */
+;(function ($)
+{
+	$.Helpfulness = function (options)
+	{
+		var documentId = $("[data-documentid]").attr("data-documentid");
+		var docId = typeof (READTHEDOCS_DATA) != 'undefined' ? READTHEDOCS_DATA.docroot + READTHEDOCS_DATA.page : options.domain != "" ? window.location.href : "";
+
+		return {
+			targetDocumentId: documentId,
+			targetPickerId: $("[data-pickerid]").attr("data-pickerid"),
+			targetDocId: docId,
+			targetKeyId: documentId || docId || 0,
+			$container: $(".helpfulness-container"),
+			$content: $(".helpfulness"),
+			$positionContainer: $("#helpful-position"),
+			shouldShow: false,
+			closed: false,
+			completed: false,
+
+			init: function ()
+			{
+				this.initHandlers();
+				this.initLoad();
+			},
+
+			initHandlers: function ()
+			{
+				var _this = this;
+
+				$("#helpfulness-btn-yes").click(function ()
+				{
+					_this.submitYes();
+				});
+
+				$("#helpfulness-btn-no").click(function ()
+				{
+					$(".helpfulness-form").hide();
+					$(".helpfulness-form-no").show();
+				});
+
+				$("#txt-helpfulness").on('input propertychange paste', function ()
+				{
+					var left = options.maxCharacters - $(this).val().length;
+					if (left < 0)
+					{
+						left = 0;
+					}
+					$("#helpfulness-characters-left").text(left + " character" + (left == 1 ? "" : "s") + " remaining");
+				});
+
+				$("#helpfulness-btn-submit").click(function ()
+				{
+					var reason = $("#txt-helpfulness").val().substring(0, 160);
```

Comment:
The max length of `160` seems to be hard-coded in lots of places and so it isn't really a proper setting. The max length option needs to be plumbed through to all the relevant locations instead of hard-coding.